code/__DEFINES/dcs/helpers.dm

```diff
@@ -0,0 +1,15 @@
+/// Used to trigger signals and call procs registered for that signal
+/// The datum hosting the signal is automaticaly added as the first argument
+/// Returns a bitfield gathered from all registered procs
+/// Arguments given here are packaged in a list and given to _SendSignal
+#define SEND_SIGNAL(target, sigtype, arguments...) ( !target.comp_lookup || !target.comp_lookup[sigtype] ? NONE : target._SendSignal(sigtype, list(target, ##arguments)) )
+
+#define SEND_GLOBAL_SIGNAL(sigtype, arguments...) ( SEND_SIGNAL(SSdcs, sigtype, ##arguments) )
+
+/// A wrapper for _AddElement that allows us to pretend we're using normal named arguments
+#define AddElement(arguments...) _AddElement(list(##arguments))
```

Comment:
It's either this or everyone calling the proc has to make the list themselves, I'm fine with either

Reply:
Not a fan of obfuscating how parameters are constructed.

Reply:
is the `##` doing anything here?

Reply:
You need it for optional arguments in macros

Reply:
it does two things:
- attach to preceding ident
- strip preceding comma

neither is relevant

Reply:
Ah right it was the commas inside the macro that it cared about, still though it's harmless and makes less changes required if I later make the type a separate argument as I probably should anyway.

Reply:
My only gripe is PascalCase defines which triggers my internal reeeeeee

Reply:
I am okay with wrapping for the developer convenience here.